plugins/npm/src/tasks/npm-publish.ts

```diff
@@ -0,0 +1,70 @@
+import { ToolKitError } from '@dotcom-tool-kit/error'
+import { Task } from '@dotcom-tool-kit/types'
+import pacote from 'pacote'
+import { readState } from '@dotcom-tool-kit/state'
+import pack from 'libnpmpack'
+import { publish } from 'libnpmpublish'
+import { styles } from '@dotcom-tool-kit/logger'
+import { readdirSync } from 'fs'
+
+export const semVerRegex = /^v\d+\.\d+\.\d+(-.+)?/
+
+export default class NpmPublish extends Task {
+  static description = ''
+  
+  handleTagValidity(tag: string): void {
+    if(!tag) {
+        throw new ToolKitError('CIRCLE_TAG environment variable not found. Make sure you are running this on a release version!')
+    }
+    if(!semVerRegex.test(tag)) {
+        throw new ToolKitError(`CIRCLE_TAG does not match regex ${semVerRegex}. Configure your release version to match the regex eg. v1.2.3-beta.8`)
+    }
+  }
+
+  async listPackedFiles(packagePath: string): Promise<void> {
+    this.logger.info('packing these files:')
+    readdirSync(packagePath).forEach(fileDir => {
+      this.logger.info(fileDir)
+    })
```

Comment:
**issue/suggestion**: this is logging every file in the folder; but that's probably not the same as what's actually being published, it's not taking into account `.npmignore`, `.gitignore`, the `files` field in `package.json` etc. this could be pretty misleading! what you could probably do instead is read the actual list of files in the generated tarball (using https://github.com/npm/node-tar/ maybe?), so we know exactly what files are being published.